-1

**Failed Tests**:   ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-e38aa6/27857/summary.html
**COMMIT**: b4f48b05c5a835b55e83ba4448a6bd4350fc4219
**CMSSW**: CMSSW_12_5_X_2022-09-29-1100/el8_amd64_gcc10
**Additional Tests**: GPU
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/39533/27857/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package DataFormats/PortableTestObjects
>> Entering Package DataFormats/SoATemplate
>> Entering Package HeterogeneousCore/AlpakaTest
>> Entering Package HeterogeneousCore/CUDATest
>> Compile sequence completed for CMSSW CMSSW_12_5_X_2022-09-29-1100
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_12_5_X_2022-09-29-1100/tmp/el8_amd64_gcc10/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package CUDADataFormats/PortableTestObjects
Entering library rule at CUDADataFormats/PortableTestObjects
>> Building LCG reflex dict from header file src/CUDADataFormats/PortableTestObjects/src/classes.h
</pre>

--- a/DataFormats/SoATemplate/interface/SoACommon.h
+++ b/DataFormats/SoATemplate/interface/SoACommon.h
@@ -390,7 +390,7 @@ namespace cms::soa {
     }
 
     using ValueType = typename C::Scalar;
-    static constexpr auto valueSize = sizeof(C::Scalar);
+    static constexpr auto valueSize = sizeof(typename C::Scalar);
     SOA_HOST_DEVICE SOA_INLINE byte_size_type stride() const { return stride_; }
 
   private:
@@ -503,7 +503,7 @@ namespace cms::soa {
     SOA_HOST_DEVICE SOA_INLINE const C* operator&() const { return &cVal_; }
 
     using ValueType = typename C::Scalar;
-    static constexpr auto valueSize = sizeof(C::Scalar);
+    static constexpr auto valueSize = sizeof(typename C::Scalar);
 
     SOA_HOST_DEVICE SOA_INLINE byte_size_type stride() const { return stride_; }
 

